sequencer.c

```diff
@@ -1005,6 +1006,17 @@ static int run_git_commit(struct repository *r,
 				setenv("GIT_COMMITTER_DATE", date, 1);
 				free(date);
 			}
+			if (opts->ignore_date) {
+				struct strbuf new_author = STRBUF_INIT;
+
+				while (*author != '>')
```

Comment:
We should check that we don't run over the end of the string here, e.g. if somebody changes the author script on disk, we still shouldn't read some random memory.

Even better I think we can replace this `while` loop with a `memchr` to get the length of the bits we need to split off, and, add that to the `strbuf` using `strbuf_add` and then `datestamp` it.

Other than being easier to read, I think this also allows us to plug a memory leak, as we are leaking the original `author` right now.  (I think we're also leaking it currently, but now we'd leak two instances)

I also wonder if it would be better to traverse this string from the end, rather than from the front.  I don't think we allow `>` in the author name, so this should work fine in all cases, but I'd still feel more comfortable going over something that isn't user controlled.

Also I think `author` could be `NULL` here right, if the `is_rebase_i` clause is not satisfied.  We should also check for that I think.

Reply:
> Even better I think we can replace this whileloop with a memchr to get the length of the bits we need to split off, and, add that to the strbuf using strbuf_add and then datestamp it.

datestamp uses user's local TZ which is in contrast with behavior of --ignore-date of am (which sets it to UTC, reading from one of the tests). Agree on everything else.

> Also I think author could be NULL here right, if the is_rebase_i clause is not satisfied. We should also check for that I think.

`--ignore-date` is only set by rebase -i so I think we are clear on this one.

Reply:
> `--ignore-date` is only set by rebase -i so I think we are clear on this one.

Ah yes of course, that makes sense.  It would be nice to have a `BUG()` statement then, indicating that we know about this, and helping developers in the future if they want to add another codepath here, e.g. if they want to start having a similar functionality in `cherry-pick`.